jcstress-samples/src/main/java/org/openjdk/jcstress/samples/concurrency/classic/ClassicProblem_02_ProducerConsumerProblem.java

```diff
@@ -0,0 +1,312 @@
+        @Actor
+        void p1(III_Result r) {
+            r.r1 = produce();
+        }
+
+        @Actor
+        void p2(III_Result r) {
+            r.r2 = produce();
+        }
+
+        @Actor
+        void c(III_Result r) {
+            r.r3 = consume();
+        }
+    }
+
+    /**
+     * The solution with semaphores can be extended so that more than one producer and consumer are supported.
+     */
+    @JCStressTest
+    @Outcome(expect = ACCEPTABLE)
+    @Outcome(id = {"0, 0, 2"}, expect = ACCEPTABLE_INTERESTING, desc = "Producers used the same index at the same time.")
+    @State
+    public static class FixedTwoProducersOneConsumer extends SemaphoresBase {
+        private final Object indexLock = new Object();
+        private int index = 0;
+
+        @Actor
+        void p1(III_Result r) {
+            r.r1 = produce();
+        }
+
+        @Actor
+        void p2(III_Result r) {
+            r.r2 = produce();
+        }
+
+        @Actor
+        void c(III_Result r) {
+            r.r3 = consume();
+        }
+
+        @Override
+        protected int putItemIntoBuffer() {
+            synchronized (indexLock) {
+                return index++;
+            }
+        }
+
+        @Override
+        protected void takeItemFromBuffer() {
+            synchronized (indexLock) {
+                index--;
+            }
+        }
+    }
+
+    /**
+     * This solution with a ReentrantLock and two conditions works with many producers and many consumers.
+     */
+    @JCStressTest
+    @Outcome(expect = ACCEPTABLE)
+    @Outcome(id = {"0, 0, 2"}, expect = ACCEPTABLE_INTERESTING, desc = "Producers used the same index at the same time.")
+    @State
+    public static class Lock {
+        private final ReentrantLock lock = new ReentrantLock();
+        private final Condition full = lock.newCondition();
+        private final Condition empty = lock.newCondition();
+        private int count = 0;
+
+        @Actor
+        void p1(III_Result r) {
+            r.r1 = produce();
+        }
+
+        @Actor
+        void p2(III_Result r) {
+            r.r2 = produce();
+        }
+
+        @Actor
+        void c(III_Result r) {
+            r.r3 = consume();
+        }
+
+        public int produce() {
+            // produce item
+            lock.lock();
+            try {
+                while(count == BUFFER_SIZE) {
+                    full.await();
+                }
+                // put item to buffer
+                int index = count++;
+                if(count == 1) {
+                    empty.signalAll();
+                }
+                return index;
+            } catch (InterruptedException e) {
+                throw new IllegalStateException(e);
+            } finally {
+                lock.unlock();
+            }
+        }
+
+        public int consume() {
+            int result;
+            lock.lock();
+            try {
+                while(count == 0) {
```

Comment:
The style is to leave space after `while`, `if`, `for`, etc.

Reply:
IntelliJ always adds those spaces when I use the auto-format feature or any code completion feature. 
I don't put them into the samples by intention. 

Have you got a code style file for me in which your favourite code style for JCStress is defined? If so, then please send it to me via email and I will use it so that you don't need to bother such stylistic issues anymore in this PR and neither in any next PR. Ok?